utils/nctl/sh/assets/setup_shared.sh

```diff
@@ -432,45 +432,99 @@ function setup_asset_node_configs()
     done
 }
 
+function validators_string()
+{
+    local IDX
+    local COUNT_NODES_AT_GENESIS=$(get_count_of_genesis_nodes)
+    local PATH_TO_NET
+
+    local VALIDATORS
+
+    PATH_TO_NET="$(get_path_to_net)"
+
+    for IDX in $(seq 1 "$COUNT_NODES_AT_GENESIS")
+    do
+        VALIDATORS+="--validator $(cat "$PATH_TO_NET/nodes/node-$IDX/keys/public_key_hex"),$(get_node_pos_stake_weight "$COUNT_NODES_AT_GENESIS" "$IDX") "
+    done
+
+    echo $VALIDATORS
+}
+
 function setup_asset_global_state_toml() {
     log "... setting node global_state.toml"
 
     local COUNT_NODES=${1}
-    local PROTOCOL_VERSION=${2}
+    local TARGET_PROTOCOL_VERSION
     local IDX
     local GLOBAL_STATE_OUTPUT
-    local PATH_TO_NET
-    local PRE_1_4_0
-
-    PATH_TO_NET="$(get_path_to_net)"
+    local GLOBAL_STATE_VALIDATOR_OUTPUT
+    local PATH_TO_NET="$(get_path_to_net)"
+    local GLOBAL_STATE_UPDATE_TOOL_MINOR_VERSION
+    local STATE_ROOT_HASH
+    local PRE_14
+    local DATA_LMDB_PATH
+
+    # Check version of the global state update tool. This is not a proper semver check, for simplicity, we rely on the minor part only.
```

Comment:
yeah, i wrote `check_next_version` and stuff like that on the `ProtocolVersion` struct back in the day in the EE and it looks like it survived in somewhat altered form in the casper_types crate in the new repo. it may or may not be useful to you, but here it is:

https://github.com/casper-network/casper-node/blob/35c5f797e3c612dff98bd5aef9709816be591546/types/src/protocol_version.rs#L74

Reply:
Admittedly, this goes a bit out of scope of this PR, but otoh it's a good call to improve version handling at least in this part of NCTL. I'll implement a bit more sophisticated version check using bash trickery.

Reply:
Semver version checking added in https://github.com/casper-network/casper-node/pull/3451/commits/0ae555fba7dbc51062dbba1ef0472cd87815d2de